### PR TITLE
[SiteTypes] Add S2 operator for S=1 and update code style

### LINKED ISSUE #1501: [ITensors] [BUG] Operator "S2" does not work for SiteType "S=1"
**Description of bug**

It is not possible to create an `MPO()` with operator `S2` for the SiteType `S=1`. `S2` is listed as an available operator for SiteType `S=1` [here](https://itensor.github.io/ITensors.jl/stable/IncludedSiteTypes.html#%22S1%22-SiteType). It should be noted that the `S2`-operator works fine for SiteType `S=1/2`.

**Minimal code demonstrating the bug or unexpected behavior**

<details><summary>Minimal runnable code</summary><p>

```julia
using ITensors
sites = siteinds("S=1",10)
os = OpSum()
os += "S2", 3
mpo = MPO(os,sites)

```
</p></details>

**Expected output or behavior**

Expected an `mpo` corresponding to operator `S2` acting on site 3.

**Actual output or behavior**

<details><summary>Output of minimal runnable code</summary><p>

```julia
ERROR: ArgumentError: Overload of "op" or "op!" functions not found for operator name "S2" and Index tags: ("S=1,Site,n=3",).
```
</p></details>

**Version information**

 - Output from `versioninfo()`:
```julia
julia> versioninfo()
Julia Version 1.10.4
Commit 48d4fd48430 (2024-06-04 10:41 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 10 × Apple M1 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)
```

 - Output from `using Pkg; Pkg.status("ITensors")`:
```julia
julia> using Pkg; Pkg.status("ITensors")
Status `~/.julia/environments/v1.10/Project.toml`
⌅ [9136182c] ITensors v0.3.68
Info Packages marked with ⌅ have new versions available but compatibility constraints restrict them from upgrading. To see why use `status --outdated`
```

### PR DESCRIPTION
**Pull request by emstoudenmire**
This PR fixes #1501 by defining the "S2" (aka "S²") operator for the "S=1" SiteType. It also updates the code for "S=1" operator definitions to use the `op` matrices approach rather than the `op!` functions. A unit test was added for the "S2" operator.

### CODE REVIEW

**Comment by emstoudenmire** [comment]
Hm, when I rerun JuliaFormatter on the code it doesn't make any changes, yet it failed the Format Check. This also happened to me another time recently. I wonder if the formatting styles are out of sync between the CI and the code repo somehow? Or maybe these are just one-offs.

**Comment by mtfishman** [comment]
What version of JuliaFormatter.jl do you have locally?

The CI formatting suggestion looks correct to me:
```julia
julia> using JuliaFormatter

julia> format_text("1."; style=DefaultStyle())
"1.0"

julia> format_text("1."; style=BlueStyle())
"1.0"

julia> import Pkg; Pkg.status("JuliaFormatter")
Status `~/.julia/environments/v1.10/Project.toml`
  [98e50ef6] JuliaFormatter v1.0.56
```
See https://github.com/domluna/JuliaFormatter.jl/pull/66.

File: src/lib/SiteTypes/src/sitetypes/spinone.jl [hunk 1]
```diff
@@ -51,103 +51,83 @@ state(::StateName"Y+", ::SiteType"S=1") = [-1 / 2, -im / sqrt(2), 1 / 2]
...
-  Op[s' => 1, s => 3] = -0.5
-  return Op[s' => 3, s => 3] = +0.5
-end
+op(::OpName"Sz", ::SiteType"S=1") = [
+  1.0 0.0 0.0
+  0.0 0.0 0.0
+  0.0 0.0 -1.0
+]
+
+op(::OpName"Sᶻ", t::SiteType"S=1") = op(OpName("Sz"), t)
```
Comment (mtfishman) [inline src/lib/SiteTypes/src/sitetypes/spinone.jl:60]: I've been using a code style like:
```suggestion
op(on::OpName"Sᶻ", t::SiteType"S=1") = op(alias(on), t)
```
and then would define an `alias(...)` overload if needed.

Comment (mtfishman) [reply src/lib/SiteTypes/src/sitetypes/spinone.jl:60]: Same thing elsewhere.

**Comment by emstoudenmire** [comment]
I resolved the formatter issue, thanks.

I didn't realize about the aliases – nice pattern.
I defined aliases for the spin operators and updated both spinone.jl and qubit.jl (underlying "S=1/2") to use these.
Also I made an alias for "SpinOne" -> "S=1" inside spinone.jl.

**Comment by emstoudenmire** [comment]
Of course one could envision many improvements to this system, like maybe a more unified way of handling spin operators since e.g. some of the aliases are repeated a bit, but I think this PR is a good "maintenance" sort of PR.

**Comment by mtfishman** [comment]
Yeah it would be good to step back and be a bit more systematic about the operator definitions code but that can be done in future work.

**Comment by mtfishman** [comment]
Thanks!

**Pull request merged**